metafuse/include/metafuse.hpp

```diff
@@ -864,22 +946,36 @@ class FuseFs
 {
 public:
 
-    int main(int argc, char const* argv[], bool default_options = true)
-    {
-        std::vector<char const*> argv_vec(argv, argv+argc);
+    template <typename ArgsT, typename OptionsT>
+    int main(ArgsT const &args
+             , OptionsT &&options
+             , bool default_options = true)
+    {
+        argv_.clear();
+        std::copy(args.cbegin(), args.cend(), std::back_inserter(argv_));
+        auto join_options = [](OptionsT const &options) {
+            std::vector<std::string> parts;
+            for (auto const &kv : options) {
+                auto const &v = kv.second;
+                parts.push_back(v.size() ? kv.first + "=" + v : kv.first);
+            }
+            return "-o" + boost::algorithm::join(parts, ",");
+        };
         if(default_options) {
             update_uid();
             update_gid();
-            //argv_vec.push_back("-s");
-            argv_vec.push_back("-o");
-            argv_vec.push_back("default_permissions");
-            argv_vec.push_back("-o");
-            argv_vec.push_back(_uid.c_str());
-            argv_vec.push_back("-o");
-            argv_vec.push_back(_gid.c_str());
+            options["uid"] = std::to_string(::getuid());
```

Comment:
yes, it is bettery if it is provided explicitely: default_permissions is not compatible with some other options on some fuse versions